relay-sampling/src/lib.rs

```diff
@@ -324,12 +341,59 @@ pub struct SamplingRule {
     #[serde(rename = "type")]
     pub ty: RuleType,
     pub id: RuleId,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub time_range: Option<TimeRange>,
```

Comment:
Is it worth skipping serialization in cases the rule has expired?

Reply:
I won't be pursuing this for now. I don't want to put the time into correctly setting the serialization skipping function, plus testing it. Ideally, there won't be many expired rules; and if the amount of non-active rules becomes an issue, we probably need to take more action than just skipping the serialization here.

Unless someone disagrees with this and considers it a blocker to landing this PR, I'll keep this as a to-do for some time in the future.

Reply:
I'm +1 on skipping serialization eventually, since that is also Sentry's behavior.

Reply:
Done in 98e15ae0137d22404454f5ca0b60ea92fc1b1cf1. That commit also supports open ranges, so serialization is only skipped when both `start` and `end` are empty.